### PR TITLE
Add `matching-brackets` exercise

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

**Review by glennj — APPROVED** [review]

**Comment by axtens** [comment]
```
ok> "erik" "" s:/ .
 ["e","r","i","k"]
ok> "erik" null s:/ .
 ["e","r","i","k"]
```
null is what's suggested by the manual, viz
```
C:\Users\bugma\Documents\GitHub>8th -? "s:/"

s:/
SED: s n -- a | s a1 -- a2 | s pat -- a | s null -- a

Split the string s at n, splitting into an array [head,tail] at that character index (starting from 0).

If an array of numbers a1 is given, splits into an array of strings with that many characters each, repeating the
last value as long as there is still something to split (e.g. "12345" [2] s:/ will give ["12","34","5]).

If string or a regex pat is given, splits on all occurrences of pat, omitting it from the results.

If null is given, splits s into its component characters.

Returns an array of results, or an array with the entire string as the only element, if there are no matches.

See also: b:/ a:join
```

File: exercises/practice/matching-brackets/.meta/example.8th [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  dup "(" s:= if r> ")" a:push >r drop recurse ;then
+  dup "{" s:= if r> "}" a:push >r drop recurse ;then
+  dup "]" s:= if r> a:len 0 n:= if 3drop false ;then a:pop swap >r s:= !if drop false ;then recurse ;then
+  dup ")" s:= if r> a:len 0 n:= if 3drop false ;then a:pop swap >r s:= !if drop false ;then recurse ;then
+  dup "}" s:= if r> a:len 0 n:= if 3drop false ;then a:pop swap >r s:= !if drop false ;then recurse ;then
+  drop recurse
+;
+
+: paired? \ s -- T
+  a:new >r "" s:/ paired!
```
Comment (ErikSchierboom) [inline exercises/practice/matching-brackets/.meta/example.8th:14]:
```suggestion
  a:new >r null s:/ paired!
```

**Comment by ErikSchierboom** [comment]
I've updated @axtens.

**Comment by axtens** [comment]
Do you also want to look at Ron's content on breaking out of an anonymous function?

**Comment by ErikSchierboom** [comment]
> Do you also want to look at Ron's content on breaking out of an anonymous function?

Yeah sure!

**Comment by ErikSchierboom** [comment]
@axtens Example solution has been updated.

**Pull request merged**